File: sdk/samples/WFSamples/Applications/OrderingStateMachine/CS/OrderLocalServices/OrderService.cs
//---------------------------------------------------------------------
//  This file is part of the Windows Workflow Foundation SDK Code Samples.
// 
//  Copyright (C) Microsoft Corporation.  All rights reserved.
// 
//This source code is intended only as a supplement to Microsoft
//Development Tools and/or on-line documentation.  See these other
//materials for detailed information regarding Microsoft code samples.
// 
//THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
//KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
//IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
//PARTICULAR PURPOSE.
//---------------------------------------------------------------------

using System;

namespace Microsoft.Samples.Workflow.OrderApplication
{
    [Serializable]
    public class OrderService : IOrderService
    {
        public void RaiseOrderCreatedEvent(string orderId,Guid instanceId)
        {
            if (OrderCreated != null)
            {
                OrderEventArgs e = new OrderEventArgs(instanceId, orderId);
                OrderCreated(this, e);
            }
        }

        public void RaiseOrderShippedEvent(string orderId,Guid instanceId)
        {
            if (OrderShipped != null)
            {
                OrderEventArgs e = new OrderEventArgs(instanceId, orderId);
                OrderShipped(this, e);
            }
        }

        public void RaiseOrderUpdatedEvent(string orderId,Guid instanceId)
        {
            if (OrderUpdated != null)
            {
                OrderEventArgs e = new OrderEventArgs(instanceId, orderId);
                OrderUpdated(this, e);
            }
        }
        public void RaiseOrderProcessedEvent(string orderId,Guid instanceId)
        {
            if (OrderProcessed != null)
            {
                OrderEventArgs e = new OrderEventArgs(instanceId, orderId);
                OrderProcessed(this, e);
            }
        }
        public void RaiseOrderCanceledEvent(string orderId,Guid instanceId)
        {
            if (OrderCanceled != null)
            {
                OrderEventArgs e = new OrderEventArgs(instanceId, orderId);
                OrderCanceled(this, e);
            }
        }
        public event EventHandler<OrderEventArgs> OrderCreated;
        public event EventHandler<OrderEventArgs> OrderShipped;
        public event EventHandler<OrderEventArgs> OrderUpdated;
        public event EventHandler<OrderEventArgs> OrderProcessed;
        public event EventHandler<OrderEventArgs> OrderCanceled;
    }

}

File: sdk/samples/WFSamples/Applications/OrderingStateMachine/VB/OrderLocalServices/OrderService.vb
'---------------------------------------------------------------------
'  This file is part of the Windows Workflow Foundation SDK Code Samples.
' 
'  Copyright (C) Microsoft Corporation.  All rights reserved.
' 
'This source code is intended only as a supplement to Microsoft
'Development Tools and/or on-line documentation.  See these other
'materials for detailed information regarding Microsoft code samples.
' 
'THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
'KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
'IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
'PARTICULAR PURPOSE.
'---------------------------------------------------------------------

Imports System

<Serializable()> _
Public Class OrderService
    Implements IOrderService
    Public Sub RaiseOrderCreatedEvent(ByVal orderId As String, ByVal instanceId As Guid)
        Dim e As New OrderEventArgs(instanceId, orderId)
        RaiseEvent OrderCreated(Me, e)
    End Sub

    Public Sub RaiseOrderShippedEvent(ByVal orderId As String, ByVal instanceId As Guid)
        Dim e As New OrderEventArgs(instanceId, orderId)
        RaiseEvent OrderShipped(Me, e)
    End Sub

    Public Sub RaiseOrderUpdatedEvent(ByVal orderId As String, ByVal instanceId As Guid)
        Dim e As New OrderEventArgs(instanceId, orderId)
        RaiseEvent OrderUpdated(Me, e)
    End Sub


    Public Sub RaiseOrderProcessedEvent(ByVal orderId As String, ByVal instanceId As Guid)
        Dim e As New OrderEventArgs(instanceId, orderId)
        RaiseEvent OrderProcessed(Me, e)
    End Sub

    Public Sub RaiseOrderCanceledEvent(ByVal orderId As String, ByVal instanceId As Guid)
        Dim e As New OrderEventArgs(instanceId, orderId)
        RaiseEvent OrderCanceled(Me, e)
    End Sub

    Public Event OrderCanceled(ByVal sender As Object, ByVal e As OrderEventArgs) Implements IOrderService.OrderCanceled

    Public Event OrderCreated(ByVal sender As Object, ByVal e As OrderEventArgs) Implements IOrderService.OrderCreated

    Public Event OrderProcessed(ByVal sender As Object, ByVal e As OrderEventArgs) Implements IOrderService.OrderProcessed

    Public Event OrderShipped(ByVal sender As Object, ByVal e As OrderEventArgs) Implements IOrderService.OrderShipped

    Public Event OrderUpdated(ByVal sender As Object, ByVal e As OrderEventArgs) Implements IOrderService.OrderUpdated

End Class
